dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -144,64 +145,82 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         return it->second;
     }
 
-    bool tryLoadSchemaDiffs(Getter & getter, Int64 version, Context & context)
+    Int64 tryLoadSchemaDiffs(Getter & getter, Int64 lastest_version, Context & context)
     {
-        if (isTooOldSchema(cur_version, version))
+        if (isTooOldSchema(cur_version, lastest_version))
         {
-            return false;
+            return -1;
         }
 
         LOG_FMT_DEBUG(log, "try load schema diffs.");
 
-        SchemaBuilder<Getter, NameMapper> builder(getter, context, databases, version);
+        SchemaBuilder<Getter, NameMapper> builder(getter, context, databases, lastest_version);
 
         Int64 used_version = cur_version;
-        std::vector<SchemaDiff> diffs;
-        while (used_version < version)
+        while (used_version < lastest_version)
```

Comment:
We should first fetch all schema diff and save them, and then apply them locally. Should not apply fetch and apply inside one while loop.
It is because that fetching schema diff from tikv is not exception-safe. If we do fetch and apply inside one while loop, we may get inconsistent schema.

For example:
The cur_version is 50 and we call tryLoadSchemaDiff with latest_version=70
Fetch and apply schema diff 51~60 successfully
When fetching schema diff 61, it throw an exception caused by network
Next time when we call tryLoadSchemaDiff, the cur_version is still 50 but the local schema copy is at schema_version=60

Fetch all schema diff and save them firstly, and then apply them locally in another loop won't cause this problem.

Reply:
How about add a try catch when call `getter.getSchemaDiff(used_version);`?
Then when 61 failed, it still can roll to 60.